planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp

```diff
@@ -1723,6 +1723,31 @@ void AvoidanceModule::generateExtendedDrivableArea(ShiftedPath * shifted_path) c
     }
   }
 
+  {
+    const bool include_opposite = parameters_.enable_avoidance_over_opposite_direction;
+    if (include_opposite) {
+      lanelet::ConstLanelets search_lanelets;
+      lanelet::ConstLanelet current_lane;
+      if (route_handler->getClosestLaneletWithinRoute(getEgoPose().pose, &current_lane)) {
+        constexpr bool get_right = true;
+        constexpr bool get_left = true;
+        const auto current_ego_shift = getCurrentShift();
+        if (fabs(current_ego_shift) > 0.01) {
+          if (current_ego_shift > 0.0) {
+            search_lanelets = route_handler->getAllSharedLineStringLanelets(
+              current_lane, !get_right, get_left, include_opposite);
+          } else {
+            search_lanelets = route_handler->getAllSharedLineStringLanelets(
+              current_lane, get_right, !get_left, include_opposite);
+          }
+          if (search_lanelets.size() > 1) {
+            extended_lanelets.push_back(search_lanelets.at(1));
```

Comment:
@shulanbushangshu 
I don't have a map to test, but technically, there is a possibility that the drivable area extension might will result with the same issue if the number of opposite lane > 1. 

Another thing of my concern in that multiple similar lanelet will be pushed back (i think the same issue already existed), and it might affect the performance a bit. 

I have another solution to this, but I will need to discuss internally about this part, and let you know about it.

Reply:
Thank you for your examination.I  use "extended_lanelets.push_back(search_lanelets.at(1)) " to ensure that the current adjacent lane is used instead of multiple lanes.I have tested the two opposite lanes and the result is well.


https://user-images.githubusercontent.com/102840938/184335777-0429403f-4411-4dc5-b807-f8e61b601d02.mp4

Reply:
@shulanbushangshu 
@zulfaqar-azmi-t4 -san says search_lanelets can be size one and that will cause segmentation fault at `extended_lanelets.push_back(search_lanelets.at(1))`

I think the point for this PR is getCurrentShift() can be more suitable than isOnRight() as you show.
@shulanbushangshu can you please consider that reason at first?